doc/man/8/mount.ceph.rst

```diff
@@ -1,182 +1,213 @@
+    to use the standard search path for ceph.conf files.
 
-:command:`osdkeepalive`
-  int, Default: 5
+:command: `mds_namespace=<fs-name>`
+      Specify the non-default file system to be mounted. Not passing this
+      option mounts the default file system.
 
 :command:`mount_timeout`
-  int (seconds), Default: 60
+    int (seconds), Default: 60
 
-:command:`osd_idle_ttl`
-  int (seconds), Default: 60
+:command:`name`
+    RADOS user to authenticate as when using CephX. Default: guest
 
-:command:`caps_wanted_delay_min`
-  int, cap release delay, Default: 5
+:command:`secret`
+    secret key for use with CephX. This option is insecure because it exposes
+    the secret on the command line. To avoid this, use the secretfile option.
 
-:command:`caps_wanted_delay_max`
-  int, cap release delay, Default: 60
+:command:`secretfile`
+    path to file containing the secret key to use with CephX
 
+:command:`recover_session=<no|clean>`
+    Set auto reconnect mode in the case where the client is blacklisted. The
+    available modes are ``no`` and ``clean``. The default is ``no``.
+
+    - ``no``: never attempt to reconnect when client detects that it has been
+       blacklisted. Blacklisted clients will not attempt to reconnect and
+       their operations will fail too.
+
+    - ``clean``: client reconnects to the Ceph cluster automatically when it
+      detects that it has been blacklisted. During reconnect, client drops
+      dirty data/metadata, invalidates page caches and writable file handles.
+      After reconnect, file locks become stale because the MDS loses track of
+      them. If an inode contains any stale file locks, read/write on the inode
+      is not allowed until applications release all stale file locks.
+
+Advanced
+--------
 :command:`cap_release_safety`
-  int, Default: calculated
+    int, Default: calculated
 
-:command:`readdir_max_entries`
-  int, Default: 1024
+:command:`caps_wanted_delay_max`
+    int, cap release delay, Default: 60
 
-:command:`readdir_max_bytes`
-  int, Default: 524288 (512*1024)
+:command:`caps_wanted_delay_min`
+    int, cap release delay, Default: 5
 
-:command:`write_congestion_kb`
-  int (kb), max writeback in flight. scale with available
-  memory. Default: calculated from available memory
+:command:`dirstat`
+    funky `cat dirname` for stats, Default: off
 
-:command:`snapdirname`
-  string, set the name of the hidden snapdir. Default: .snap
+:command:`ip`
+    my ip
 
-:command:`name`
-  RADOS user to authenticate as when using cephx. Default: guest
+:command:`noasyncreaddir`
+    no dcache readdir
 
-:command:`secret`
-  secret key for use with cephx. This option is insecure because it exposes
-  the secret on the command line. To avoid this, use the secretfile option.
+:command:`nocrc`
+    no data crc on writes
 
-:command:`secretfile`
-  path to file containing the secret key to use with cephx
+:command:`nodirstat`
+    no funky `cat dirname` for stats
```

Comment:
I wonder if we ought to not document the "no*" options separately, and instead just document that some options can be negated by prefixing them with "no".

Reply:
That's a nice suggestion. I've lumped such options together.

Right now, there's no options like `asyncreaddir`, `crc` and `share` described in man page but there are options like `noasyncreaddir`, `nocrc` and `noshare`. Should I add them? grepping for these options in `src/` yielded nothing.